### ISSUE TITLE
Is the webpack assets supposed to be in git or not?

### ISSUE THREAD

**Issue by alfredvaa** [shakacode/react_on_rails#903]
When using Heroku, the assets generated with assets:precompile should be added to git. Is it the same for the webpack assets in public/webpack/production? 

It seems to me that the webpack assets does not exist when running the app from heroku, but the /public/webpack/* is on the other hand in .gitignore which implies that they should not be added...

How is it supposed to work?

**Comment by justin808** [comment]
> When using Heroku, the assets generated with assets:precompile should be added to git. 

Where is this documented? I believe the precompiled assets *should **never** be added to git*.

> Is it the same for the webpack assets in public/webpack/production?
See my previous statement

> It seems to me that the webpack assets does not exist when running the app from heroku, but the /public/webpack/* is on the other hand in .gitignore which implies that they should not be added...

`.gitignore` is for developers so git knows that stuff in the deployment directories is not a **pending** change.

> How is it supposed to work?
Have I answered your questions above?

If you want personalized help on this, please consider our [Coaching Plan](http://www.shakacode.com/work/shakacode-coaching-plan.pdf).

**Comment by alfredvaa** [comment]
@justin808 It is in the Heroku dokumentation: https://devcenter.heroku.com/articles/rails-asset-pipeline _"Adding public/assets to your git repository will make it available to Heroku..."_

I don't need a Coaching Plan, but it would be great with a complete documentation, this part seems to be out of date: https://github.com/shakacode/react_on_rails/blob/master/docs/additional-reading/rails-assets-relative-paths.md, since I am using v8+...

**Comment by justin808** [comment]
@alfredvaa The Heroku docs are an inferior way.

> Note: this doc needs updating to reflect how v8.x+ no longer puts Webpack generated files through the asset pipeline. PR's welcome!

We'd love a PR to help!

**Comment by alfredvaa** [comment]
I guess that I kind of solved this for now. Instead of adding the precompiled assets to git, I let Heroku manage the precompilation. But for this to work there should not be a sprocets-manifest file in the public/assets folder added to git, so I had to remove that first (this should probably be added to .gitignore). I also had to set up buildpacks to use one for nodejs and one for ruby, but that is already in the docs for react on rails.
